addons/payment_adyen/controllers/main.py

```diff
@@ -135,6 +139,7 @@ def adyen_payments(
                 f'/payment/adyen/return?merchantReference={reference}'
             ),
         }
+        data.update(tx_sudo._adyen_extend_api_data(data))
```

Comment:
It doesn't work because `data` is not specified yet.
We could remove `data` parameter from the method call, but
I'd prefer more flexible implementation (see my comment about `_adyen_extend_api_data`)

Reply:
I meant to apply this suggestion *after* removing the `data` parameter.